src/server/cluster/cluster_family.cc

```diff
@@ -689,6 +696,41 @@ void ClusterFamily::DflySlotMigrationStatus(CmdArgList args, ConnectionContext*
   return rb->SendSimpleString(state_to_str(MigrationState::C_NO_STATE));
 }
 
+void ClusterFamily::DflyClusterMigrationFinalize(CmdArgList args, ConnectionContext* cntx) {
+  CmdArgParser parser{args};
+  auto sync_id = parser.Next<uint32_t>();
+
+  if (auto err = parser.Error(); err) {
+    return cntx->SendError(err->MakeReply());
+  }
+
+  auto migration_it = outgoing_migration_jobs_.find(sync_id);
+  if (migration_it == outgoing_migration_jobs_.end()) {
+    return cntx->SendError(absl::StrCat("finalize invalid session id ", sync_id));
+  }
```

Comment:
we erase at the bottom, what if we have two outgoing migrations that finalize at the same time (their connections are placed on different threads)?